x/upgrade/plan/info.go

```diff
@@ -99,7 +104,12 @@ func (m BinaryDownloadURLMap) CheckURLs(daemonName string) error {
 		return fmt.Errorf("could not create temp directory: %w", err)
 	}
 	defer os.RemoveAll(tempDir)
-	for osArch, url := range m {
+
+	// Ensure that downloads proceed in a deterministic pattern.
+	osArches := maps.Keys(m)
+	sort.Strings(osArches)
```

Comment:
Nope, we need these deterministic.

Reply:
Surprising! But if `map.Keys`  doesn't sort the keys, then the dependency provides very little value, right? It would be better to define your own `package map` with a `Keys` that returned ordered keys, I guess?

Reply:
Is that requirement documented somewhere?